be/src/column/column.h

```diff
@@ -275,6 +276,15 @@ class Column {
     // FIXME: Many derived implementation assume |to| equals to size().
     virtual size_t filter_range(const Filter& filter, size_t from, size_t to) = 0;
 
+    // Sort this column incrementally, and build tie for the next column
+    // @param is_asc_order ascending order or descending order
+    // @param is_null_first null first or null last
+    // @param permutation input and output permutation
+    // @param tie input and output tie
+    // @param range sort range, {0, 0} means not build tie but sort data
+    virtual void sort_and_tie(const bool& cancel, bool is_asc_order, bool is_null_first, SmallPermutation& permutation,
```

Comment:
This method only used by sort operator，So I think would better use ColumnVisitor here.

Reply:
So the design principle is concentrating data structure in the `Column`, and separating specific algorithms into other module?

I generally agree with this, except for a little concern is that `Column` needs to expose some low-level interface for data access, which may break the abstraction of `Column` interface.

Anyway, I could take a try.